CMakeLists.txt

```diff
@@ -102,6 +102,14 @@ else()
   target_link_libraries(mlc_chat_cli PUBLIC mlc_llm)
 endif()
 
+add_library(mlc_llm_module MODULE $<TARGET_OBJECTS:mlc_llm_objs>)
```

Comment:
Let us first avoid  it in windows, since that might be problematic, and only add this target in linux mac for now

Reply:
Addressed in https://github.com/mlc-ai/mlc-llm/pull/223/commits/17277a394e7aeca54c2dbc6faef706ab1b8c201f.